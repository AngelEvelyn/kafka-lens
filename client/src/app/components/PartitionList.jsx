import React from 'react';
import Partition from '../components/Partition.jsx';

const PartitionList = props => {
    let partitionsArray = []
    const numberOfPartitions = props.topicInfo.partition;

    for (let i = 0; i < numberOfPartitions; i++) {
        partitionsArray.push(<Partition key={i} id={i} showMessages={props.showMessages} topicName={props.topicInfo.topic} />)
    }

  return <div>{partitionsArray}</div>;
};

export default PartitionList;