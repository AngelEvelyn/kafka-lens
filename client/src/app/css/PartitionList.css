.partition-list {
    align-self: center;
    justify-self: center;
    width: 75%;
    
}