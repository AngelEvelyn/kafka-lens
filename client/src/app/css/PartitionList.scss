@import 'theme';

.partition-list {
    display: flex;
    flex-direction: column;
    align-items: center;
}
