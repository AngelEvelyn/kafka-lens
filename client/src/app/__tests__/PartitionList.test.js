import React from 'react';
import { shallow } from 'enzyme';

import PartitionList from '../components/PartitionList';

describe('ParitionList unit tests', () => {
  let wrapper;
  const props = {
    topicInfo: {
      partition: 8
    }
  };
  beforeAll(() => {
    wrapper = shallow(<PartitionList {...props} />);
  });

  it('Should render without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render a Partition component', () => {
    expect(wrapper.exists('Partition')).toBe(true);
  });
});
