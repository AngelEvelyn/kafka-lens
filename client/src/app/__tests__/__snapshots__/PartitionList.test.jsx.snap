// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartitionList unit tests Should render without errors 1`] = `
<div
  className="partition-list"
>
  <Partition
    id="0"
    key="first-0"
    showMessages={[MockFunction]}
    topicName="first"
  />
  <Partition
    id="1"
    key="first-1"
    showMessages={[MockFunction]}
    topicName="first"
  />
  <Partition
    id="2"
    key="first-2"
    showMessages={[MockFunction]}
    topicName="first"
  />
  <Partition
    id="3"
    key="first-3"
    showMessages={[MockFunction]}
    topicName="first"
  />
  <Partition
    id="4"
    key="first-4"
    showMessages={[MockFunction]}
    topicName="first"
  />
</div>
`;
